addon-test-support/@ember/test-helpers/settled.ts

```diff
@@ -142,12 +143,52 @@ function checkWaiters() {
   return false;
 }
 
+interface IPauseTestPromiseDetails {
+  label: undefined | string;
+  error: Error;
+}
+
+let currentPauseTestPromises = new Map<Promise<unknown>, IPauseTestPromiseDetails>();
+
+/**
+  Ensures that tests wait for the provided promise before being considered "settled".
+
+  This function is expected to be used in application/addon code (aka non-test code).
```

Comment:
I am a little worried about this for several reasons:
- apps and addons don't have an explicit dependency on `@ember/test-helpers` currently
- would addons that use this need to declare it in `dependencies`?
- should it be a peer dependency instead?
- how would that work if apps only have it as a transitive dependency via e.g. `ember-qunit`?
- what if an addon declares it as a dependency, another addon declares it as a dependency with different version and those conflict?

I'm wondering if it would be better to have this in a separate addon until these bundling/dependency issues are resolved 🤔

Reply:
Yes, I agree that some of this becomes easier in a separate addon, however the ability to debug/track becomes harder. Specifically, the only way a standalone addon could make the pausing behavior work would be via a test waiter. With a classic test waiter situation we would have absolutely no information about _why_ waiters are not settled.